programs/bpf/rust/stack_bug_dep/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+//! @brief Example Rust-based BPF program tests loop iteration
+
+#![no_std]
+#![allow(unused_attributes)]
+
+extern crate alloc;
+#[cfg(not(test))]
+extern crate solana_sdk_bpf_no_std;
+extern crate solana_sdk_bpf_utils;
+
+use alloc::vec::Vec;
+use solana_sdk_bpf_utils::info;
+use solana_sdk_bpf_utils::entrypoint::{SolPubkey};
+
+pub struct InitPollData<'a> {
+    pub timeout: u32, // block height
+    pub header_len: u32,
+    pub header: &'a [u8],
+    pub option_a_len: u32,
+    pub option_a: &'a [u8],
+    pub option_b_len: u32,
+    pub option_b: &'a [u8],
+}
+
+pub struct PollData<'a> {
+    pub creator_key: &'a SolPubkey,
+    pub last_block: u64,
+    pub header_len: u32,
+    pub header: &'a [u8],
+    pub option_a: PollOptionData<'a>,
+    pub option_b: PollOptionData<'a>,
+}
+
+impl<'a> PollData<'a> {
+    pub fn length(&self) -> usize {
+        (32 + 8 + 4 + self.header_len) as usize + self.option_a.length() + self.option_b.length()
+    }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
+        info!(self.length(), self.header_len, self.option_a.length(), 0, 0);
+        let mut bytes = Vec::with_capacity(self.length());
+        bytes.extend_from_slice(self.creator_key);
+        bytes.extend_from_slice(&self.last_block.to_be_bytes());
+        bytes.extend_from_slice(&self.header_len.to_be_bytes());
+        bytes.extend_from_slice(self.header);
+        info!("option_a to_bytes");
+        bytes.extend(self.option_a.to_bytes().into_iter());
+        info!("option_b to_bytes");
+        bytes.extend(self.option_b.to_bytes().into_iter());
+        info!("return poll data");
+        bytes
+    }
+
+    // TODO error checking
+    pub fn init(
+        init: InitPollData<'a>,
+        creator_key: &'a SolPubkey,
+        tally_a_key: &'a SolPubkey,
+        tally_b_key: &'a SolPubkey,
+        slot: u64,
+    ) -> Self {
+        assert_eq!(init.timeout, 10);
```

Comment:
But this fails

Reply:
Just this assert, or others too?

Reply:
Just below this assertion, there are a few more which also would fail if they were to run. They are redundant, I'll remove